File: BlotoutAnalytics/Models/BOACaptureModel.m
//
//  BOCaptureModel.m
//  BlotoutAnalytics
//
//  Copyright © 2021 Blotout. All rights reserved.
//

#import "BOACaptureModel.h"

@implementation BOACaptureModel

- (instancetype)initWithEvent:(NSString *)event
                   properties:(NSDictionary *)properties screenName:(NSString* _Nullable)screenName withType:(NSString*)type {
  if (self = [super init]) {
    _event = event;
    _properties = properties != nil ? properties : @{};
    _screenName = screenName != nil ? screenName : @"";
    _type = type;
  }
  
  return self;
}

@end

File: BlotoutAnalytics/Models/BOACaptureModel.swift
//
//  BOACaptureModel.swift
//  BlotoutAnalyticsSDK
//
//  Created by Poonam Tiwari on 21/03/22.
//

import Foundation
class BOACaptureModel: NSObject {
    var event = ""
    var properties: [AnyHashable : AnyHashable]?
    var eventSubCode: NSNumber?
    var screenName: String?
    var type: String?

    init(
        event: String,
        properties: [AnyHashable : AnyHashable]?,
        screenName: String?,
        withType type: String?
    ) {
        super.init()
            self.event = event
            self.properties = properties ?? [:]
            self.screenName = screenName ?? ""
            self.type = type
    }
}
